src/librustc/middle/resolve_lifetime.rs

```diff
@@ -992,11 +992,14 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
             let unsubst = if let Some(id) = self.hir_map.as_local_node_id(def_id) {
                 &map.object_lifetime_defaults[&id]
             } else {
-                let cstore = &self.sess.cstore;
                 self.xcrate_object_lifetime_defaults.entry(def_id).or_insert_with(|| {
-                    cstore.item_generics_cloned(def_id).types.into_iter().map(|def| {
-                        def.object_lifetime_default
-                    }).collect()
+                    ty::tls::with_opt(|opt_tcx| {
```

Comment:
Hmm, yes, the reason you are having this problem is that the `tcx` is not constructed yet. In general, fetching the tcx out of TLS is frowned upon, except for very limited cases (e.g., in a `Debug` impl) where there is basically no way to thread it down manually.

For now, you might want to leave those methods that are in code without access to a tcx -- but really the right answer is to convert these computations into "on-demand computations" themselves (e.g., resolve-lifetimes probably wants to become a query).